src/utils/DataSourceHelper/DataSourceHelper.ts

```diff
@@ -0,0 +1,91 @@
+import "./ui/DataSourceHelper.scss";
+
+interface ConstraintStore {
+    [widgetId: string]: string | HybridConstraint;
+}
+
+interface Version {
+    major: number; minor: number; path: number;
+}
+
+interface HybridConstraint {
+    attribute: string;
+    operator: string;
+    value: string;
+    path?: string;
+}
+
+interface ListView extends mxui.widget._WidgetBase {
+    _datasource: {
+        _constraints: HybridConstraint[] | string;
+        _pageObjs: mendix.lib.MxObject[];
+    };
+    datasource: {
+        type: "microflow" | "entityPath" | "database" | "xpath";
+    };
+    update: (obj: mendix.lib.MxObject | null, callback?: () => void) => void;
+    _entity: string;
+}
+
+export class DataSourceHelper {
+    static VERSION: Version = { major: 1, minor: 0, path: 0 };
+    version: Version;
+    private delay = 100; // TODO check optimal timing.
```

Comment:
Did you checked what the time could be 
User performance settings chrome to test.

Reply:
Was handled in pairing-session sometime ago